streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamPeek.java

```diff
@@ -35,10 +39,35 @@ public KStreamPeek(final ForeachAction<K, V> action) {
     }
 
     private class KStreamPeekProcessor extends AbstractProcessor<K, V> {
+        
+        @Override
+        public void init(ProcessorContext context) {
+            super.init(context);
+            if (action instanceof PrintForeachAction) {
+                ((PrintForeachAction) action).setContext(context);
```

Comment:
or we can check and set in `init()` and make a variable to record it whether should load default serde in `apply()`, that is what you think @mjsax ?